scripts/eval/rag_evaluator/ragas_metrics.py

```
@@ -0,0 +1,395 @@
+        assert sample.retrieved_contexts is not None, "Retrieved Context is not set"
+
+        if (sample.response.strip() == "") or ("\n".join(sample.retrieved_contexts).strip().strip() == ""):
+            return 0.
+        if sample.response.strip() == "\n".join(sample.retrieved_contexts).strip():
+            return 1.
+        if sample.response.strip() in "\n".join(sample.retrieved_contexts).strip():
+            return 1.
+
+        try:
+            score0 = score1 = np.nan
+            for retry in range(5):
```

Ruff (0.15.10)

[warning] Loop control variable `retry` not used within loop body

Rename unused `retry` to `_retry`

(B007)